### PR TITLE
CSRF and CSP

### LINKED ISSUE #14: Common web security practicies
* [x] CSRF via [gorilla/csrf](github.com/gorilla/csrf/)
* [x] Content Security Policy (CSP)
* [ ] Cross-Origin Resource Sharing (CORS)

### PR DESCRIPTION
**Pull request by cryptix**
Done:

* [x] CSRF
* [x]  TLS redirects and sanity checks
* [x] HSTS

fixes #14

### CODE REVIEW

File: web/handlers/http.go [hunk 1]
```diff
@@ -152,11 +154,26 @@ func New(
 		auth.SetStore(store),
 		auth.SetErrorHandler(authErrH),
 		auth.SetNotAuthorizedHandler(notAuthorizedH),
+		auth.SetLifetime(2*time.Hour), // TODO: configure
```
Comment (cryptix) [inline web/handlers/http.go:157]: @staltz this was also why we were seeing these short session times.

open to better defaults.

Comment (staltz) [reply web/handlers/http.go:157]: Makes sense!

**Comment by cryptix** [comment]
Should be good to merge but I want to check a couple of things (like the HSTS implications) once my brain is less foggy.

**Comment by cryptix** [comment]
Okay.. I just checked and  realized that `default-src` means no inline styling.

**Comment by cryptix** [comment]
Also need to fix mime-types on assets.

**Pull request merged**